### PR TITLE
CORE-1547 graphz for blockchain

### PR DESCRIPTION
**Pull request by rabbitonweb**
## Overview
Introduces new module `graphz` that allows building grphs in DOT language with a small memory footprint (potentially big graphs).

This PR creates just the module (tests are showing the usage). Next PR will use this module to build DAG representation from Casper.

### JIRA ticket:
CORE-1547

### CODE REVIEW

File: graphz/src/test/scala/coop/rchain/graphz/GraphzSpec.scala [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+  implicit class GraphzOps(graph: Effect[Graphz[Effect]]) {
+    def show: String =
+      graph.runS(new StringBuffer).toString
+
+    import java.io.File
+    import java.io.PrintWriter
+
+    def view(): Unit = {
+      val sourcePath = "/Users/rabbit/temp.gv"
+      val outputPath = "/Users/rabbit/output.pdf"
```
Comment (adaszko) [inline graphz/src/test/scala/coop/rchain/graphz/GraphzSpec.scala]: This shouldn't be here I presume

File: graphz/src/test/scala/coop/rchain/graphz/GraphzSpec.scala [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+      new File(sourcePath).createNewFile()
+      val writer = new PrintWriter(sourcePath)
+      writer.println(show)
+      writer.flush()
+      writer.close
+      val dotCmd  = s"dot -Tpdf $sourcePath -o $outputPath"
```
Comment (adaszko) [inline graphz/src/test/scala/coop/rchain/graphz/GraphzSpec.scala]: What if `$sourcePath` has spaces in it?

File: graphz/src/test/scala/coop/rchain/graphz/GraphzSpec.scala [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+      val openCmd = s"open $outputPath"
```
Comment (adaszko) [inline graphz/src/test/scala/coop/rchain/graphz/GraphzSpec.scala]: ditto

File: graphz/src/test/scala/coop/rchain/graphz/GraphzSpec.scala (hunk 2 above)
Comment (rabbitonweb) [reply graphz/src/test/scala/coop/rchain/graphz/GraphzSpec.scala]: this is just in tests for debugging purposes - this whole method will be removed

File: graphz/src/test/scala/coop/rchain/graphz/GraphzSpec.scala (hunk 1 above)
Comment (rabbitonweb) [reply graphz/src/test/scala/coop/rchain/graphz/GraphzSpec.scala]: this is for debug, will be removed eventually

Comment (lukasz-golebiewski) [reply graphz/src/test/scala/coop/rchain/graphz/GraphzSpec.scala]: When?

**Pull request merged**